pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/io/stream/ReadableStreamingData.java

```diff
@@ -130,8 +132,97 @@ public long skip(final long n) {
             long numSkipped = in.skip(clamped);
             position += numSkipped;
             return numSkipped;
-        } catch (IOException e) {
+        } catch (final IOException e) {
+            throw new DataAccessException(e);
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public long readBytes(@NonNull final byte[] dst, final int offset, final int maxLength) {
+        if (maxLength < 0) {
+            throw new IllegalArgumentException("Negative maxLength not allowed");
+        }
+        if (!hasRemaining()) {
+            return 0;
+        }
+        try {
+            int bytesRead = in.read(dst, offset, maxLength);
+            position += bytesRead;
+            if (bytesRead < maxLength) {
```

Comment:
This should also check that bytesRead is less than `dst.length` so we do not try to write past the end of the array.